### ISSUE TITLE
Segfault when resizing window in scratchpad

### ISSUE THREAD

**Issue by lbonn** [swaywm/sway#1389]
sway version 0.15-rc1-1-ga82aa2a (2017-10-10, branch "master")

Test case:

```
urxvt -T test& sleep 1; swaymsg '[title="test"] floating enable, move scratchpad, resize set 100px 100px'
```

Backtrace:

```
Program received signal SIGSEGV, Segmentation fault.
update_geometry (container=0x5555561b47f0) at sway/sway/layout.c:753
753             swayc_t *op = workspace->parent;
(gdb) bt
#0  0x0000555555584feb in update_geometry (container=0x5555561b47f0) at sway/sway/layout.c:753
#1  0x0000555555589f35 in update_geometry (container=container@entry=0x5555561b47f0)
    at sway/sway/layout.c:931
#2  0x000055555556b277 in set_size_floating (use_width=<optimized out>, new_dimension=<optimized out>)
    at sway/commands/resize.c:33
#3  0x000055555556b277 in set_size (dimension=<optimized out>, use_width=<optimized out>)
    at sway/sway/commands/resize.c:218
#4  0x000055555556b97f in cmd_resize_set (argv=0x5555561b84f0, argc=<optimized out>)
    at sway/sway/commands/resize.c:319
#5  0x000055555556b97f in cmd_resize (argc=<optimized out>, argv=0x5555561b84e8)
    at sway/sway/commands/resize.c:332
#6  0x00005555555639a2 in handle_command (_exec=_exec@entry=0x5555561ba750 "[title=test] floating enable, move scratchpad, resize set 100px 100px", context=context@entry=CONTEXT_IPC) at sway/sway/commands.c:471
#7  0x0000555555583d9c in ipc_client_handle_command (client=0x5555561c2170)
    at sway/sway/ipc-server.c:703
#8  0x00005555555841db in ipc_client_handle_command (client=0x5555561c2170)
    at sway/sway/ipc-server.c:672
#9  0x00005555555841db in ipc_client_handle_readable (client_fd=30, mask=<optimized out>, data=0x5555561c2170)
    at sway/sway/ipc-server.c:257
#10 0x00007ffff72c59b2 in wl_event_loop_dispatch () at /usr/lib/libwayland-server.so.0
#11 0x00007ffff72c414c in wl_display_run () at /usr/lib/libwayland-server.so.0
#12 0x00007ffff7bb3f4d in wlc_run () at /usr/lib/libwlc.so.0
#13 0x0000555555562455 in main (argc=<optimized out>, argv=<optimized out>) at sway/sway/main.c:464
```

Log (excerpt): https://gist.github.com/lbonn/441e8868fbd8176350296457b80eb634

**Comment by SirCmpwn** [comment]
Man, our scratchpad implementation really sucks. I should really rewrite it.

**Comment by SirCmpwn** [comment]
I'm closing issues that are resolved by the yet-unreleased sway 1.0.
